### DISCUSSION TITLE
Ajax call on search event

### DISCUSSION THREAD

**Discussion by Malin88** [TarekRaafat/autoComplete.js#251] [Q&A]
Hi,
I set `threshold` to `2` and on begin my data src is empty. How can I fire ajax call to fetch data based on typed query ?

**Comment by folknor** [comment]
I'm fairly sure I don't understand the question. Can you rephrase it?

If I understood you correctly, then this is how:
```js
data: {
	src: async (query) => {
		let data
		await fetch(url + "&q=" + query, pm)
			.then(foo)
			.then(json => (data = json))
			.catch(bar)
		return data.baz
	},
}
```
